DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Utility.cs

```diff
@@ -0,0 +1,172 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.DeID.SharedLib.Model;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core
+{
+    public class Utility
+    {
+        public static DateTimeOffset[] ParseDicomDate(DicomDate item)
+        {
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            return item.Get<string[]>().Select(ParseDicomDate).ToArray();
+        }
+
+        public static DateTimeOffset ParseDicomDate(string date)
+        {
+            EnsureArg.IsNotNull(date, nameof(date));
+
+            return DateTimeOffset.ParseExact(date, "yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static DateTimeObject[] ParseDicomDateTime(DicomDateTime item)
+        {
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            return item.Get<string[]>().Select(ParseDicomDateTime).ToArray();
+        }
+
+        public static DateTimeObject ParseDicomDateTime(string dateTime)
+        {
+            EnsureArg.IsNotNull(dateTime, nameof(dateTime));
+
+            Regex dateTimeRegex = new Regex(@"^((?<year>\d{4})(?<month>\d{2})(?<day>\d{2})(?<hour>\d{2})(?<minute>\d{2})(?<second>\d{2})(\.(?<millisecond>\d{1,6}))?(?<timeZone>(?<sign>-|\+)(?<timeZoneHour>\d{2})(?<timeZoneMinute>\d{2}))?)(\s*)");
+            var matches = dateTimeRegex.Matches(dateTime);
+            if (matches.Count != 1)
+            {
+                throw new Exception();
+            }
+
+            var groups = matches[0].Groups;
+
+            int year = groups["year"].Success ? int.Parse(groups["year"].Value) : 0;
```

Comment:
>0; [](http://example.com/codeflow?start=82&length=2)

is it possible to set this value to 0, and will a zero bring some troubles to you in the following processing work? should we just simply throw exception if Success is false?

Reply:
Some parts are optional. e.g. millisecond.
Set value to 0 will not effect the results.
Of course throw exception in some fields is reasonable and more strict.